### PR TITLE
Move the edit links outside of the <h#> elements

### LINKED ISSUE #1966: Move Edit links out of headings
Right now the edit links are located within the heading (for example `<h1>`) elements. Besides being semantically incorrect, this leads to to "Edit" occasionally being considered part of the heading.

### PR DESCRIPTION
**Pull request by gtback**
- Use flexbox rather than absolute positioning.
- Give headers a consistent top and bottom margin.

Fix #1966

### CODE REVIEW

**Comment by gtback** [comment]
Once the docs-build runs, I'll put some before-and-after screenshots. I don't think there will be much significant visual difference.

File: resources/web/style/heading.pcss [hunk 1]
```diff
@@ -1,4 +1,23 @@
...
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    /* Give all headings plenty of space above them */
+    margin-top: 40px;
+    margin-bottom: 10px;
+
+    a, h1, h2, h3, h4, h5, h6 {
+      display: flex
+    }
```
Comment (JoshMock) [inline resources/web/style/heading.pcss:13]: What is the benefit of employing flexbox here? It's within a pretty small scope so I guess introducing accidental flexbox bugs is probably unlikely but I'm just curious. I've learned just enough flexbox to be equal parts dangerous and wary of its ways. 😄

Comment (gtback) [reply resources/web/style/heading.pcss:13]: It's the best way **I know** to have two elements center-aligned vertically, while having the second one pushed all the way to the right. That's not the same as it being the _actual_ best way, though. 😁 

I started trying by trying to set the `.edit_me` icons as `float: right`, but the vertical alignment was a mess, and I was trying to minimize the changes to HTML (I  basically just added the `heading` class to the enclosing `<div>` to make it easier to target with this rule.

Comment (JoshMock) [reply resources/web/style/heading.pcss:13]: Totally fair. `display: flex` is the new `float:` 😄

**Review by JoshMock — APPROVED** [review]

**Comment by gtback** [comment]
I still need to do some work on this. The `-40px` doesn't work correctly in all cases (on some pages, it's not needed).

**Comment by gtback** [comment]
I spot-checked a bunch of pages, and they are all looking pretty consistent to me. I'd appreciate if some @elastic/docs writers could take a look at pages in [this preview](https://docs_1967.docs-preview.app.elstc.co/guide/index.html) to make sure you all don't notice anything weird. 

I did notice that the headline on [this page](https://docs_1967.docs-preview.app.elstc.co/guide/en/beats/libbeat/7.9/whats-new.html) isn't getting wrapped correctly. I'll make sure to fix that before merging.

**Comment by bmorelli25** [comment]
I wonder if a top margin or padding should be added? For example, the H3s on [this page](https://docs_1967.docs-preview.app.elstc.co/guide/en/observability/current/ingest-logs.html) seem tight:

---

Screen Shot 2020-11-05 at 11 39 08 AM

---

Screen Shot 2020-11-05 at 11 39 18 AM

---

Screen Shot 2020-11-05 at 11 39 27 AM

**Comment by lcawl** [comment]
There seems to be something strange happening here:
https://docs_1967.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index.html

image

**Comment by gtback** [comment]
Thanks for these examples, @bmorelli25 and @lcawl. I'll take a closer look. I'm curious why the headings are so inconsistent throughout the site 🤷

**Comment by gtback** [comment]
It looks like the issue is that the `titlepage` div is used in multiple places on the page :-( I'll see if I can make that CSS exception more specific.

**Comment by gtback** [comment]
I've spent too long on this already, and the HTML that gets generated is inconsistent enough that there's not an elegant CSS solution I've found without a ton of edge cases which I'd like to avoid. I'm just going to close this, but leave the branch in case this becomes a priority again.

**Pull request closed without merging**